src/backend/commands/dropcmds.c

```diff
@@ -109,8 +109,11 @@ RemoveObjects(DropStmt *stmt)
 
 		/* Check permissions. */
 		namespaceId = get_object_namespace(&address);
-		if (!OidIsValid(namespaceId) ||
-			!object_ownercheck(NamespaceRelationId, namespaceId, GetUserId()))
+		if ((!OidIsValid(namespaceId) ||
+			!object_ownercheck(NamespaceRelationId, namespaceId, GetUserId())) &&
+			!(OidIsValid(namespaceId) && IS_BBF_DB_DDLADMIN(namespaceId)) &&
+			!(stmt->removeType == OBJECT_SCHEMA && IS_BBF_DB_DDLADMIN(address.objectId)) &&
+			!(relation != NULL && IS_BBF_DB_DDLADMIN(RelationGetNamespace(relation))))
```

Comment:
1. Was it necessary to have three condition?
2. Why last condition is necessary -- Shouldn't the namespaceId will already have relation's namespace?

Reply:
I can explain the three conditions
1. general case when we have a schema contained object and schema OID is available.
2. DROP SCHEMA in this case namespaceId is NULL and we need to get the schema OID from address.
3. Triggers are table level objects and for them namespaceId is always NULL, since pg_trigger does not have a column for namespace. So instead we look at the relation's namespaceId